### PR TITLE
2078 docs for approx functions

### PR DESCRIPTION
**Pull request by jychen7**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/2078

 # Rationale for this change

Add user guide for aggregate functions, similar to https://trino.io/docs/current/functions/aggregate.html

# What changes are included in this PR?

# Are there any user-facing changes?

only Documentation

### CODE REVIEW

**Comment by jychen7** [comment]
@alamb , original issue was to add user guide for approximate functions, but seems we are also lack of aggregate functions (general, approximate, statistical)

if you have time, can take early review of `general` and `approximate` aggregate functions documentation, I will add `statistical` soon

**Comment by matthewmturner** [comment]
Im very interested in this as well and was planning on looking into it as part of overall doc efforts i planned on doing with #1927 / #1821.  i havent had chance to start looking into it but do you know if there was a way to have the function definitions link to the relevant parts of the code where the functions are defined?

And this could probably be done as a follow on PR but i always found value in seeing examples in function docs (for example https://prestodb.io/docs/current/functions/json.html)

File: docs/source/user-guide/sql/aggregate_functions.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+it supports raw data as input and build Tdigest sketches during query time.
+
+### approx_percentile_cont_with_weight
+`approx_percentile_cont_with_weight(x, w, p) -> x` return the approximate percentile (TDigest) of input values with weight, where `w` is weight column expression and `p` is a float64 between 0 and 1 (inclusive).
+
+it supports raw data as input or pre-aggregated TDigest sketches (mean and weight), then build and merge Tdigest sketches during query time.
+it is suitable for low latency query OLAP system where `Spark Streaming/Flink` pre-aggregate data to a data store, then query by Datafusion.
+
+## Statistical
```
Comment (xudong963) [inline docs/source/user-guide/sql/aggregate_functions.md:52]: WIP?

File: docs/source/user-guide/sql/sql_status.md [hunk 2]
```diff
@@ -76,6 +76,9 @@
   - [x] nullif
 - Approximation functions
   - [x] approx_distinct
+  - [x] approx_median
```
Comment (xudong963) [inline docs/source/user-guide/sql/sql_status.md:79]: 👍🏻

**Review by xudong963 — COMMENTED** [review]
Thanks @jychen7

File: docs/source/user-guide/sql/aggregate_functions.md (hunk 1 above)
Comment (jychen7) [reply docs/source/user-guide/sql/aggregate_functions.md:52]: yes, user guide of statistical aggregate functions is still wip.
let me change PR to draft first

File: docs/source/user-guide/sql/aggregate_functions.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
+# Aggregate Functions
+
+Aggregate functions operate on a set of values to compute a single result.
+
+## General
+
+### min
```
Comment (alamb) [inline docs/source/user-guide/sql/aggregate_functions.md:7]: I think writing nice / proper documentation for SQL is a large undertaking.

For standard SQL functions, perhaps we can simply link to pre-existing documentation, such as: https://www.postgresql.org/docs/current/functions-aggregate.html and focus DataFusion documentation on what is different / specific to datafusion (such as`approx_percentile_cont`, `approx_median` and `approx_percentile_cont_with_weight`)?

File: docs/source/user-guide/sql/sql_status.md (hunk 2 above)
Comment (alamb) [reply docs/source/user-guide/sql/sql_status.md:79]: 💯  thank you

File: docs/source/user-guide/sql/aggregate_functions.md (hunk 3 above)
Comment (jychen7) [reply docs/source/user-guide/sql/aggregate_functions.md]: good point, I add the PostgreSQL link and only list what we support for General and Statistical functions. Approximate functions keep same detail documentations

[bcba321](https://github.com/apache/arrow-datafusion/pull/2082/commits/bcba321e0706f6ab1369b4ac730d13930f06bc6c)

File: docs/source/user-guide/sql/aggregate_functions.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+## Statistical
+
+- var / var_samp / var_pop
+- stddev / stddev_samp / stddev_pop
+- covar / covar_samp / covar_pop
+- corr
+
+## Approximate
+
+### approx_distinct
```
Comment (alamb) [inline docs/source/user-guide/sql/aggregate_functions.md:42]: 👍

File: docs/source/user-guide/sql/aggregate_functions.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+`approx_percentile_cont(x, p) -> x` return the approximate percentile (TDigest) of input values, where `p` is a float64 between 0 and 1 (inclusive).
+
+it supports raw data as input and build Tdigest sketches during query time.
+
+it is approximately equal to `approx_percentile_cont_with_weight(x, 1, p)`.
+
+### approx_percentile_cont_with_weight
+
+`approx_percentile_cont_with_weight(x, w, p) -> x` return the approximate percentile (TDigest) of input values with weight, where `w` is weight column expression and `p` is a float64 between 0 and 1 (inclusive).
```
Comment (alamb) [inline docs/source/user-guide/sql/aggregate_functions.md:60]:
```suggestion
`approx_percentile_cont_with_weight(x, w, p) -> x` returns the approximate percentile (TDigest) of input values with weight, where `w` is weight column expression and `p` is a float64 between 0 and 1 (inclusive).
```

File: docs/source/user-guide/sql/aggregate_functions.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+`approx_median(x) -> x` returns the approximate median of input values. it is an alias of `approx_percentile_cont(x, 0.5)`.
+
+### approx_percentile_cont
+
+`approx_percentile_cont(x, p) -> x` return the approximate percentile (TDigest) of input values, where `p` is a float64 between 0 and 1 (inclusive).
+
+it supports raw data as input and build Tdigest sketches during query time.
+
+it is approximately equal to `approx_percentile_cont_with_weight(x, 1, p)`.
```
Comment (alamb) [inline docs/source/user-guide/sql/aggregate_functions.md:56]:
```suggestion
It supports raw data as input and build Tdigest sketches during query time, and is approximately equal to `approx_percentile_cont_with_weight(x, 1, p)`.
```

File: docs/source/user-guide/sql/aggregate_functions.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+### approx_percentile_cont_with_weight
+
+`approx_percentile_cont_with_weight(x, w, p) -> x` return the approximate percentile (TDigest) of input values with weight, where `w` is weight column expression and `p` is a float64 between 0 and 1 (inclusive).
+
+it supports raw data as input or pre-aggregated TDigest sketches, then build or merge Tdigest sketches during query time. TDigest sketches are a list of centroid `(x, w)`, where `x` stands for mean and `w` stands for weight.
```
Comment (alamb) [inline docs/source/user-guide/sql/aggregate_functions.md:62]:
```suggestion
It supports raw data as input or pre-aggregated TDigest sketches, then builds or merges Tdigest sketches during query time. TDigest sketches are a list of centroid `(x, w)`, where `x` stands for mean and `w` stands for weight.
```

File: docs/source/user-guide/sql/aggregate_functions.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+it is suitable for low latency OLAP system where streaming compute engine (e.g. Spark Streaming/Flink) pre-aggregate data to a data store, then query by Datafusion.
```
Comment (alamb) [inline docs/source/user-guide/sql/aggregate_functions.md:64]:
```suggestion
It is suitable for low latency OLAP system where a streaming compute engine (e.g. Spark Streaming/Flink) pre-aggregates data to a data store, then queries using Datafusion.
```

**Review by alamb — APPROVED** [review]
Thanks @jychen7  ! I left some minor suggestions but overall I think this looks great

File: docs/source/user-guide/sql/aggregate_functions.md (hunk 4 above)
Comment (jychen7) [reply docs/source/user-guide/sql/aggregate_functions.md:42]: thank you for your kindly review as well 😄

**Review by houqp — APPROVED** [review]

**Pull request merged**